openadmet/models/architecture/gat.py

```diff
@@ -0,0 +1,650 @@
+                    model_config["input_dim"] = 8  # Default molecular feature dimension
+                else:
+                    raise ValueError("input_dim must be specified or inferrable from training data")
+
+            self.estimator = GATv2LightningWrapper(
+                model_config=model_config,
+                loss_fn_name=self.loss_function,
+                lr=self.lr,
+                weight_decay=self.weight_decay,
+                scheduler_name=self.scheduler,
+                warmup_epochs=self.warmup_epochs
+            )
+
+            logger.info(f"Built GATv2LightningWrapper with GATv2Model config: {model_config}")
+            logger.info(f"LightningWrapper params: lr={self.lr}, loss={self.loss_function}, scheduler={self.scheduler}")
+        else:
+            logger.warning("Model already exists, skipping build")
+
+    def train(self, dataloader):
+        """
+        Train the model
+        """
+        raise NotImplementedError(
+            "Training not implemented in model class, use a trainer"
+        )
+
+    def predict(self, dataloader, accelerator="gpu", devices=1) -> np.ndarray:
+        """
+        Use model for prediction
+
+        Args:
+            dataloader: PyTorch Geometric DataLoader
+            accelerator: Accelerator type ("gpu", "cpu")
+            devices: Number of devices
+
+        Returns:
+            Numpy array of predictions
+        """
+        if not self.estimator:
+            raise AttributeError("Model not built or trained")
+
+        with torch.inference_mode():
+            trainer = pl.Trainer(
+                logger=None,
+                enable_progress_bar=False,
+                accelerator=accelerator,
+                devices=devices
+            )
+            preds = trainer.predict(self.estimator, dataloader)
+
+        # Return predictions as 2D array (samples, 1) to match evaluator expectations
+        preds_array = torch.cat(preds).cpu().numpy()
+        if preds_array.ndim == 1:
+            preds_array = preds_array.reshape(-1, 1)
+        return preds_array
+
+    def get_model_summary(self):
+        """
+        Get model summary information
+        """
+        if not self.estimator:
+            return "Model not built"
+
+        total_params = sum(p.numel() for p in self.estimator.parameters())
+        trainable_params = sum(p.numel() for p in self.estimator.parameters() if p.requires_grad)
+
+        summary = {
+            "model_type": "GATv2 (Graph Attention Network v2)",
+            "total_parameters": total_params,
+            "trainable_parameters": trainable_params,
+            "num_layers": self.num_layers,
+            "num_attention_heads": self.num_heads,
+            "hidden_dimension": self.hidden_dim,
+            "pooling_method": self.pooling,
+            "dropout_rate": self.gat_dropout
+        }
+
+        return summary
+
+
+def test_gat_from_yaml():
```

Comment:
Move this to the test area `openadmet/models/tests/anvil`. You can also add files to help with testing.